spec/features/user_can_login_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.feature "Login", type: :feature do
+  scenario "allows a User to login" do
+    visit '/'
+    expect(page).to have_content("You are a Guest, please login")
+    click_link "Sign Up"
+    fill_in "Username", with: "CharmingSnail"
+    fill_in "Password", with: "Pa55w0rd"
+    click_button "Submit"
+    click_link "Logout"
+    expect(page).to have_content("You are a Guest, please login")
+    click_link "Login"
+    fill_in "Username", with: "CharmingSnail"
+    fill_in "Password", with: "Pa55w0rd"
+    click_on "Login"
+    expect(page).to have_content("You are logged in as CharmingSnail")
```

Comment:
If the default is that users are automatically logged-in after they sign-up, it might be good to also include this line between lines 10 and 11; if the default is that they are not automatically logged in, then line 11 might cause an issue (because - I think - the logout button wouldn't appear if no user is logged in).

Reply:
Agreed. I would have to logout first in this case to test the login itself